### PR TITLE
API - Writing Assignment

### PR DESCRIPTION
**Pull request by robmadden**

### CODE REVIEW

File: app/controllers/api/v1/topics_controller.rb [hunk 1]
```diff
@@ -33,6 +33,18 @@ def create
     end
   end
 
+  def create_post
+    @topic = Topic.find(params[:id])
```
Comment (bmneely) [inline app/controllers/api/v1/topics_controller.rb]: Doesn't need to be @

File: app/controllers/api/v1/topics_controller.rb [hunk 2]
```diff
@@ -33,6 +33,18 @@ def create
...
+    post = @topic.posts.build(post_params)
+    post.user = @current_user
+
+    if post.save
+      render json: post.to_json, status: 200
+    else
+      render json: {error: "Post update failed", status: 400}, status: 400
```
Comment (bmneely) [inline app/controllers/api/v1/topics_controller.rb]: Post create failed.

File: app/controllers/api/v1/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Api::V1::CommentsController < Api::V1::BaseController
+  before_filter :authenticate_user, except: [:index, :show]
+  before_filter :authorize_user, except: [:index, :show]
```
Comment (bmneely) [inline app/controllers/api/v1/comments_controller.rb]: If we're going to except both actions, we probably don't need these filters.

File: spec/api/v1/controllers/comments_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Api::V1::CommentsController < Api::V1::BaseController
+  before_filter :authenticate_user, except: [:index, :show]
+  before_filter :authorize_user, except: [:index, :show]
+
+  def index
+    comments = Comment.all
+    render json: comments.to_json, status: 200
```
Comment (bmneely) [inline spec/api/v1/controllers/comments_controller_spec.rb]: Do we want any other specs here?

File: spec/api/v1/controllers/comments_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class Api::V1::CommentsController < Api::V1::BaseController
+  before_filter :authenticate_user, except: [:index, :show]
+  before_filter :authorize_user, except: [:index, :show]
+
+  def index
+    comments = Comment.all
```
Comment (bmneely) [inline spec/api/v1/controllers/comments_controller_spec.rb]: This willy be an empty array

File: spec/api/v1/controllers/comments_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class Api::V1::CommentsController < Api::V1::BaseController
```
Comment (bmneely) [inline spec/api/v1/controllers/comments_controller_spec.rb]: I think this is the wrong content.

**Comment by bmneely** [comment]
Back to you @robmadden

File: spec/api/v1/controllers/comments_controller_spec.rb (hunk 6 above)
Comment (robmadden) [inline spec/api/v1/controllers/comments_controller_spec.rb]: Yes, yes it is.

**Comment by robmadden** [comment]
@bmneely Updated. Take another look if you want or just close.

**Pull request closed without merging**